admin/class-taxonomy-presenter.php

```diff
@@ -0,0 +1,317 @@
+<?php
+/**
+ * @package WPSEO\Admin
+ */
+
+/**
+ * Class WPSEO_Taxonomy_Presenter
+ */
+class WPSEO_Taxonomy_Presenter {
+
+	/**
+	 * @var array
+	 */
+	private $options = array();
+
+	/**
+	 * @var bool|mixed
+	 */
+	private $tax_meta;
+
+	/**
+	 * @var stdClass
+	 */
+	private $term;
+
+	/**
+	 * @var array   Options array for the no-index options, including translated labels
+	 */
+	private $no_index_options = array();
+
+	/**
+	 * @var array   Options array for the sitemap_include options, including translated labels
+	 */
+	private $sitemap_include_options = array();
+
+	/**
+	 * @param stdClass $term
+	 */
+	public function __construct( $term ) {
+		$this->term     = $term;
+		$this->tax_meta = WPSEO_Taxonomy_Meta::get_term_meta( (int) $term->term_id, $term->taxonomy );
+		$this->options  = WPSEO_Options::get_all();
+
+		$this->translate_meta_options();
+	}
+
+	/**
+	 * Displaying the form fields
+	 *
+	 * @param array $fields
+	 */
+	public function display_fields( array $fields ) {
+		foreach ( $fields as $field_name => $field_options ) {
+			if ( empty( $field_options['hide'] ) ) {
+				$this->form_row( 'wpseo_' . $field_name, $field_options['label'], $field_options['description'], $field_options['type'], $field_options['options'] );
+			}
+		}
+	}
+
+	/**
+	 * Getting the fields for the general tab
+	 *
+	 * @return array
+	 */
+	public function general_fields() {
+
+		$current = 'index';
+		if ( isset( $options[ 'noindex-tax-' . $this->term->taxonomy ] ) && $this->options[ 'noindex-tax-' . $this->term->taxonomy ] === true ) {
+			$current = 'noindex';
+		}
+
+		$noindex_options['options']            = $this->no_index_options;
+		$noindex_options['options']['default'] = sprintf( $noindex_options['options']['default'], $this->term->taxonomy, $current );
+
+		if ( '0' === get_option( 'blog_public' ) ) {
+			$noindex_options['description'] = '<br /><span class="error-message">' . esc_html__( 'Warning: even though you can set the meta robots setting here, the entire site is set to noindex in the sitewide privacy settings, so these settings won\'t have an effect.', 'wordpress-seo' ) . '</span>';
+		}
+
+		$fields = array(
+			'title' => array(
```

Comment:
Perhaps it's wise to create different functions for getting the title, desc, metakey, canonical, bctitle, noindex and sitemap_include. This increases readability and will make the class easier to test.